game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java

```diff
@@ -74,6 +77,9 @@ public void addHiddenPlayerName(final String name) {
     hiddenPlayers.add(name);
   }
 
+  /**
+   * Sets the chat whose players will be displayed in this panel.
```

Comment:
Updating the `this.chat` reference may not necessarily update the UI, right? The `repaint` call would not reconstruct the UI, I don't think there is anything to really instruct the panel to redraw the `Chat`. We add various listeners though, so I suspect this always works.

Beyond that, should this comment also note the other effects, that we swap around the listeners? (and also we have a call to updatePlayerList).